dataflux_pytorch/benchmark/checkpointing/simulated/README.md

```diff
@@ -0,0 +1,115 @@
+# Benchmarking PyTorch Lightning Checkpoints with Google Cloud Storage
+
+*   This benchmarking script allows you to run and benchmark the performance of the [Fully Sharded Data Parallel (FSDP)](https://lightning.ai/docs/pytorch/stable/api/lightning.pytorch.strategies.FSDPStrategy.html) save/load function in order to save and restore checkpoints from GCS. 
+
+*   This is a simulated version of the code that runs on CPU to test the performance of GCS as a backing store for AI/ML workloads without GPU.
+
+*   The script generates [state_dict](https://pytorch.org/tutorials/recipes/recipes/what_is_state_dict.html#what-is-a-state-dict-in-pytorch) that represents the state of the model. Each node then saves a part of the state_dict to GCS (the state_dict is divided among the nodes, such that the size of each shard is the same). During restore, each node reads its shard from GCS in parallel. The `.metadata` file, which is written and read by the Rank_0 (coordinator) node, contains information about which shard each node should read.
+
+*   The official implementation of FSDP in PyTorch Lightning for save_checkpoint can be found [here]
+(https://github.com/Lightning-AI/pytorch-lightning/blob/3627c5bfac704d44c0d055a2cdf6f3f9e3f9e8c1/src/lightning/fabric/strategies/fsdp.py#L419)
```

Comment:
This link doesn't show up correctly on the preview I think because it's split across lines

Reply:
Done